src/pages/PremiumVideoCall/state/usePasscodeAuth/usePasscodeAuth.ts

```diff
@@ -16,41 +17,29 @@ export function getPasscode() {
 }
 
 export function fetchToken(
+    instantVisioToken: string,
     name: string,
     room: string,
     passcode: string,
     create_room = true
 ) {
-    return fetch(`/token`, {
-        method: 'POST',
-        headers: {
-            'content-type': 'application/json',
-        },
-        body: JSON.stringify({
-            user_identity: name,
-            room_name: room,
-            passcode,
-            create_room,
-        }),
-    })
+    const api = new Api(instantVisioToken)
+    return api
+        .createRoom()
+        .then((response) => {
+            return response.roomId
+        })
+        .then((roomId) => {
+            return api.joinRoom(roomId)
+        })
+        .then((response) => {
+            const { jwtAccessToken } = response
+            return jwtAccessToken
+        })
```

Comment:
This code will change in order to only use `joinRoom`, once we merge @HugoGresse PR. I guess I can replace with `async/await` at that moment